File: src/main/java/me/sullivan/glasslang/parser/nodes/VariableNode.java
package me.sullivan.glasslang.parser.nodes;

import me.sullivan.glasslang.interpreter.Interpreter;
import me.sullivan.glasslang.interpreter.primitives.Primitive;
import me.sullivan.glasslang.lexer.token.Token;

public class VariableNode extends Node
{

    public VariableNode(Token token)
    {
        super(token);
    }

    @Override
    public String getValue()
    {
        return token.getValue();
    }

    @Override
    public String toString()
    {
        return "Variable(" + getValue() + ")";
    }

    @Override
    public Primitive<?> visitor(Interpreter interpreter)
    {
        return interpreter.visitVariableNode(this);
    }
}

File: src/me/sullivan/glasslang/parser/nodes/VariableNode.kt
package me.sullivan.glasslang.parser.nodes

import me.sullivan.glasslang.interpreter.Interpreter
import me.sullivan.glasslang.interpreter.primitives.Primitive
import me.sullivan.glasslang.lexer.token.Token

class VariableNode(token: Token) : Node(token)
{
    override val value: String
        get() = token.value

    override fun toString(): String
    {
        return "Variable($value)"
    }

    override fun visitor(interpreter: Interpreter): Primitive<*>
    {
        return interpreter.visitVariableNode(this)
    }
}
